[WIP] Argo CD getting started guide. (#30615)
openshift/openshift-docs: modules/gitops-keycloak-identity-brokering-with-openshift-oauthclient.adoc

// Module is included in the following assemblies:
//
// * 

[id="keycloak-identity-brokering-with-openshift-oauthclient_{context}"]
= Keycloak Identity Brokering with OpenShift

> This section describes how to configure a Keycloak instance to use OpenShift for authentication through Identity Brokering. This allows for Single Sign On (SSO) between the OpenShift cluster and the Keycloak instance. 

Reviewer: ```suggestion
You can configure a Keycloak instance to use OpenShift for authentication through Identity Brokering. This allows for Single Sign-On (SSO) between the OpenShift cluster and the Keycloak instance. 
```

.Prerequisites

* `jq` CLI tool is installed.

.Procedure

. Obtain the {product-title} API URL:
+
[source,terminal]
----
$ curl -s -k -H "Authorization: Bearer $(oc whoami -t)" https://<openshift-user-facing-api-url>/apis/config.openshift.io/v1/infrastructures/cluster | jq ".status.apiServerURL". 
----
+
[NOTE]
====
The address of the {product-title} API is often protected by HTTPS. Therefore, you must configure X509_CA_BUNDLE in the container and set it to `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`. Otherwise, Keycloak is not be able to communicate with the API Server.
====

> . In the Keycloak server dashboard, navigate to Identity Providers and select *Openshift v4*. Specify the following values:

Reviewer 2: Identity Providers should be in bold if it is a GUI element

*Base Url*:: OpenShift 4 API URL
*Client ID*:: `keycloak-broker`

> *Client Secret*:: any secret that you want to set in Step 7

Reviewer: ```suggestion
*Client Secret*:: Any secret that you want to set in Step 7
```

> *Client Secret*:: any secret that you want to set in Step 7.

Reviewer 3: maybe no full stop after 7?
